ui/src/features/components/TestForm/additionalInfo.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import TitleInput from '../../../components/TitleInput';
+import BodyEditor from './BodyEditor';
+import { CONTENT_TYPES } from './constants';
+import UiSwitcher from '../../../../src/components/UiSwitcher';
+
+export default class AdditionalInfo extends React.Component {
+  constructor (props) {
+    super(props);
+    const { additionalInfo } = props.scenario;
+    this.state = {
+      isActive: additionalInfo && Object.keys(additionalInfo).length > 0
+    }
+  };
+
+  onEnableDisable = (value) => {
+    const { scenario, onChangeValue } = this.props;
+    const { additionalInfo, additionalInfoLastBody } = scenario;
```

Comment:
When I started I was with your approach, but the component state can be changed to other scenario.
Let's assume there are scenario A (which contains additional info: { hi: "hello" }, and scenario B (without additional info)
1. The user disable additional info for scenario A (the component last body's state is now { hi: "hello" } )
2. The user choose to display scenario B.
3. The user enable additional info for scenario B, now the component state display { hi: "hello" } since this was the last body of the component.
Hope I explained myself correctly and understandably :)